package/kedro_viz/models/graph.py

```diff
@@ -585,6 +597,20 @@ def __post_init__(self, data_node: DataNode):
             with dataset._fs.open(load_path, **dataset._fs_open_args_load) as fs_file:
                 self.plot = json.load(fs_file)
 
+        if data_node.is_image_node():
+            from kedro.extras.datasets.matplotlib.matplotlib_writer import (
+                MatplotlibWriter,
+            )
+
+            dataset = cast(MatplotlibWriter, dataset)
+            if not dataset._exists():
+                return
+
+            load_path = get_filepath_str(dataset._get_load_path(), dataset._protocol)
+            with open(load_path, "rb") as img_file:
+                b64_string = base64.b64encode(img_file.read())
+                self.image = b64_string.decode("utf-8")

```

Comment:
```suggestion
            self.image = b64_string.decode("utf-8")
```
Since `img_file` is not used here, it's neatest to leave it outside the context manager (`with ...`).

Reply:
Noted.